torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -3043,6 +3043,146 @@ std::tuple<Tensor, Tensor, Tensor> batchnorm_double_backward(
 
 }
 
+// Helper for layernorm_double_backward
+// sum across normalized_shape (inner dimensions)
+Tensor sum_inner(const Tensor& to_sum, int axis, bool keepdim=true) {
+  auto r = to_sum;
+  for (int64_t dim = r.dim() - 1; dim >= axis; dim--) {
+    r = r.sum(dim, keepdim);
+  }
+  return r;
+}
+
+// Helper for batchnorm_double_backward
+// sum across batch size (outer dimensions)
+Tensor sum_outer(const Tensor& to_sum, int axis, bool keepdim=true) {
+  auto r = to_sum;
+  for (int64_t pos = 0; pos < axis; pos++) {
+    r = r.sum((keepdim) ? pos : 0, keepdim);
+  }
+  return r;
+}
+
+// Helper for batchnorm_double_backward
+// sum across normalized_shape (inner dimensions)
+Tensor unsqueeze_inner(const Tensor& src, int ndim) {
+  auto src_expanded = src;
+  while (src_expanded.sizes().size() < ndim) {
+    src_expanded = src_expanded.unsqueeze(1);
+  }
+  return src_expanded;
+}
+
+std::tuple<Tensor, Tensor, Tensor, Tensor> layernorm_double_backward(
```

Comment:
nit: layer_norm to be consistent with other functions

Reply:
Will do